core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -654,12 +658,17 @@ public CreatePathResult createPath(RpcContext rpcContext, LockedInodePath inodeP
           // The final path component already exists and is not persisted, so it should be added
           // to the non-persisted Inodes of traversalResult.
           syncPersistDirectory(rpcContext, (InodeDirectory) lastInode);
+
         } else if (!lastInode.isDirectory() || !(options instanceof CreateDirectoryOptions
             && ((CreateDirectoryOptions) options).isAllowExists())) {
           String errorMessage = ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path);
           LOG.error(errorMessage);
+          extensibleInodePath.getLockList().unlockLast();
           throw new FileAlreadyExistsException(errorMessage);
         }
+        // We need to remove the last inode from the locklist because it is an inode
```

Comment:
can you move this up to https://github.com/Alluxio/alluxio/pull/7342/files#diff-3ae46d8ad34ec27f13f732e3155d2d0eR655 so it is done for all code paths when the inode already exists?

Also, can you comment why it was already locked (it was locked during traversal, and locked again here).